pkg/apic/apiservice.go

```diff
@@ -2,69 +2,110 @@ package apic
-			Tags:             c.mapToTagsArray(serviceBody.Tags),
+			Attributes:       serviceBody.ServiceAttributes,
+			Tags:             mapToTagsArray(serviceBody.Tags, c.cfg.GetTagsToPublish()),
 		},
-		Spec:  c.buildAPIServiceSpec(serviceBody),
-		Owner: c.getOwnerObject(serviceBody, true),
+		Spec:  buildAPIServiceSpec(serviceBody),
+		Owner: ownerObject,
 	}
+
+	svcDetails := buildAgentDetailsSubResource(serviceBody, true, serviceBody.ServiceAgentDetails)
+	util.SetAgentDetails(svc, svcDetails)
+	c.updateAPIServerStatus(ownerObject, svc, ownerErr)
+
+	return svc
 }
 
-func (c *ServiceClient) getOwnerObject(serviceBody *ServiceBody, warning bool) *v1.Owner {
+func (c *ServiceClient) getOwnerObject(serviceBody *ServiceBody, warning bool) (*v1.Owner, error) {
 	if id, found := c.getTeamFromCache(serviceBody.TeamName); found {
 		return &v1.Owner{
 			Type: v1.TeamOwner,
 			ID:   id,
-		}
+		}, nil
 	} else if warning {
 		// warning is only true when creating service, revision and instance will not print it
-		log.Warnf("A team named %s does not exist on Amplify, not setting an owner of the API Service for %s", serviceBody.TeamName, serviceBody.APIName)
+		warnMsg := fmt.Sprintf("A team named %s does not exist on Amplify, not setting an owner of the API Service for %s", serviceBody.TeamName, serviceBody.APIName)
+		log.Warnf(warnMsg)
+		return nil, errors.New(warnMsg)
 	}
-	return nil
+	return nil, nil
 }
 
-func (c *ServiceClient) updateAPIServiceResource(apiSvc *v1alpha1.APIService, serviceBody *ServiceBody) {
-	apiSvc.ResourceMeta.Metadata.ResourceVersion = ""
-	apiSvc.Title = serviceBody.NameToPush
-	apiSvc.ResourceMeta.Attributes = c.buildAPIResourceAttributes(serviceBody, apiSvc.ResourceMeta.Attributes, true)
-	apiSvc.ResourceMeta.Tags = c.mapToTagsArray(serviceBody.Tags)
-	apiSvc.Spec.Description = serviceBody.Description
-	apiSvc.Owner = c.getOwnerObject(serviceBody, true)
+func (c *ServiceClient) updateAPIService(serviceBody *ServiceBody, svc *mv1a.APIService) {
+	var ownerErr error
+	svc.GroupVersionKind = mv1a.APIServiceGVK()
+	svc.Metadata.ResourceVersion = ""
+	svc.Title = serviceBody.NameToPush
+	svc.Tags = mapToTagsArray(serviceBody.Tags, c.cfg.GetTagsToPublish())
+	svc.Spec.Description = serviceBody.Description
+	svc.Owner, ownerErr = c.getOwnerObject(serviceBody, true)
+	svc.Attributes = serviceBody.ServiceAttributes
+
+	svcDetails := buildAgentDetailsSubResource(serviceBody, true, serviceBody.ServiceAgentDetails)
+	util.SetAgentDetails(svc, svcDetails)
+
 	if serviceBody.Image != "" {
-		apiSvc.Spec.Icon = v1alpha1.ApiServiceSpecIcon{
+		svc.Spec.Icon = mv1a.ApiServiceSpecIcon{
 			ContentType: serviceBody.ImageContentType,
 			Data:        serviceBody.Image,
 		}
 	}
+
+	c.updateAPIServerStatus(svc.Owner, svc, ownerErr)
+
+}
+
+// updateAPIServerStatus - set the subresource if team/org mismatched on API
+func (c *ServiceClient) updateAPIServerStatus(ownerObject *v1.Owner, svc *mv1a.APIService, ownerErr error) {
```

Comment:
Since this method doesn't depend on the ServiceClient, can you make it a func and find another place to put it?